### PR TITLE
Setup path correctly after stream parsing a key:value object pair

### LINKED ISSUE #2378: jq: src/jv_parse.c:305: stream_token: Assertion `k == JV_KIND_NULL' failed.
**Describe the bug**
 Reachable Assertion

**To Reproduce**
jq --stream '' input.json, the input.json is as fllow:
   {    "" : null ,-87.0E+008,
 "."
 ""
           null

jq: src/jv_parse.c:305: stream_token: Assertion `k == JV_KIND_NULL' failed.

Program received signal SIGABRT, Aborted.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff71bb859 in __GI_abort () at abort.c:79
#2  0x00007ffff71bb729 in __assert_fail_base (fmt=0x7ffff7351588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x55555569cda0 "k == JV_KIND_NULL", file=0x55555569c140 "src/jv_parse.c", line=305, function=<optimized out>) at assert.c:92
#3  0x00007ffff71ccf36 in __GI___assert_fail (assertion=assertion@entry=0x55555569cda0 "k == JV_KIND_NULL", file=file@entry=0x55555569c140 "src/jv_parse.c", line=line@entry=305, function=function@entry=0x55555569d3c0 <__PRETTY_FUNCTION__.3911> "stream_token") at assert.c:101
#4  0x00005555555dd9c0 in stream_token (ch=<optimized out>, p=<optimized out>) at src/jv_parse.c:305
#5  scan (out=<optimized out>, ch=<optimized out>, p=<optimized out>) at src/jv_parse.c:658
#6  jv_parser_next (p=<optimized out>) at src/jv_parse.c:767
#7  0x00005555555ef52d in jq_util_input_next_input (state=0x621000000100) at src/util.c:434
#8  0x000055555557908f in main (argc=argc@entry=4, argv=argv@entry=0x7fffffffe2d8) at src/main.c:676
#9  0x00007ffff71bd0b3 in __libc_start_main (main=0x555555575ac0 <main>, argc=4, argv=0x7fffffffe2d8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffe2c8) at ../csu/libc-start.c:308
#10 0x000055555557d34e in _start () at src/main.c:173

**Expected behavior**
An error, not an assert.

**Environment (please complete the following information):**
 - OS and Version:ubuntu/Linux
 - jq version [jq-stable-773baf93-dirty]

**Additional context**
Found via afl++

### PR DESCRIPTION
**Pull request by wader**
Fixes #2378 (assert on parsing invalid object)

### CODE REVIEW

**Comment by wader** [comment]
@itchyny do you think that make sense? can't claim i understand the parsing code yet

BTW do you have any 1.7 labeled issues you want help with? i've picked at few a random to look at

**Comment by itchyny** [comment]
I'll take a look at this (please forgive me for a few days).

**Comment by wader** [comment]
@itchyny no worries

File: src/jv_parse.c [hunk 1]
```diff
@@ -299,7 +299,7 @@ static pfunc stream_token(struct jv_parser* p, char ch) {
         p->output = JV_ARRAY(jv_copy(p->path), p->next);
         p->next = jv_invalid();
       }
-      p->path = jv_array_set(p->path, p->stacklen - 1, jv_true()); // ready for another name:value pair
+      p->path = jv_array_set(p->path, p->stacklen - 1, jv_null()); // ready for another name:value pair
```
Comment (itchyny) [inline src/jv_parse.c]: In the comment, `key:value pair` is better?

**Review by itchyny — APPROVED** [review]
LGTM!

File: src/jv_parse.c (hunk 1 above)
Comment (wader) [reply src/jv_parse.c]: Yeap updated and it seems to be to only place in the code where "name:value" was used

**Comment by wader** [comment]
Rebased and changed to key:value

**Comment by wader** [comment]
@itchyny I noticed that "CI / dist" seems to run for PR:s, is missing `if: startsWith(github.event.ref, 'refs/tags/jq-')`. I wonder if there is some way to restructure the workflows so we don't need if:s and can relay on `on ` instead?

**Review by itchyny — APPROVED** [review]

**Comment by wader** [comment]
Merge? Sorry dont have merge access

**Comment by itchyny** [comment]
I'll merge this, thank you. Yeah, I'll skip the dist job in #2652.

**Comment by wader** [comment]
👍

**Pull request merged**